internal/consumer/partitionConsumer.go

```diff
@@ -256,6 +256,19 @@ func (p *partitionConsumer) messageLoop(offsetRange *kafka.OffsetRange) {
 	}
 }
 
+// delayMsg try to postpone the consumption of the msg for topic.Delay duration if the current timestamp
+// is too soon compare to the msg.Timestamp when Topic.Delay is not zero.
+func (p *partitionConsumer) delayMsg(timestamp time.Time) {
+	delay := p.topicPartition.Delay
+	// check non-zero msg delay
+	if delay > 0 {
+		lag := time.Now().Sub(timestamp)
+		p.logger.Debug("delay msg deliver for topic", zap.String("topic", p.topicPartition.Name),
+			zap.Duration("sleep delay", delay-lag))
```

Comment:
It's possible `delay-lag` might be negative, although `time.Sleep` for negative time has no effect

Reply:
yes, I used to do the positive check of delay - lag. I think it's ok to pass a negative time duration to sleep since its func doc mentions that it will return immediately for negative duration.